src/components/Jobs/MonitorJobs/MonitorJobs.js

```diff
@@ -124,7 +124,7 @@ const MonitorJobs = ({
   )
 
   const pageData = useMemo(
-    () => generatePageData(handleFetchJobLogs, selectedJob),
+    () => generatePageData(handleFetchJobLogs, selectedJob, appStore, handleMonitoring),
```

Comment:
pass appStore.frontendSpec.jobs_dashboard_url and add this to dependency array, and add handleMonitoring as well